components/eamxx/src/physics/tms/eamxx_tms_process_interface.cpp

```diff
@@ -108,14 +108,20 @@ void TurbulentMountainStress::run_impl (const double /* dt */)
     // Calculate z_mid
     PF::calculate_dz(team, pseudo_density_i, p_mid_i, T_mid_i, qv_i, dz_i);
     const Real z_surf = 0.0; // For now, set z_int(i,nlevs) = z_surf = 0
+    team.team_barrier();
```

Comment:
Ah, good catch. I remember have to search for a bug in the shoc interface because no team barrier here..